common/utils/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -145,8 +148,10 @@ private[spark] class SparkDriverExecutionException(cause: Throwable)
  * Exception thrown when the main user code is run as a child process (e.g. pyspark) and we want
  * the parent SparkSubmit process to exit with the same exit code.
  */
-private[spark] case class SparkUserAppException(exitCode: Int)
-  extends SparkException(s"User application exited with $exitCode")
+private[spark] case class SparkUserAppException(exitCode: Int, errorMsg: Option[String] = None)
+  extends SparkException(
+    log"User application exited with ${MDC(LogKeys.EXIT_CODE, exitCode)}".message +
+      errorMsg.map(error => s" and caused by\n$error").getOrElse(""))
```

Comment:
I'm not super familiar with MDC this is going to use a generic exit code mapping to determine what went wrong from the exit code?